worlds/stardew_valley/logic/skill_logic.py

```diff
@@ -70,21 +65,35 @@ def can_earn_level(self, skill: str, level: int) -> StardewRule:
             xp_rule = xp_rule & self.logic.region.can_reach(Region.mines_floor_5)
         elif skill in all_mod_skills:
             # Ideal solution would be to add a logic registry, but I'm too lazy.
-            return previous_level_rule & months_rule & self.logic.mod.skill.can_earn_mod_skill_level(skill, level)
+            return previous_level_rule & self.logic.mod.skill.can_earn_mod_skill_level(skill, level)
         else:
             raise Exception(f"Unknown skill: {skill}")
 
-        return previous_level_rule & months_rule & xp_rule
+        return previous_level_rule & xp_rule
 
     # Should be cached
     def has_level(self, skill: str, level: int) -> StardewRule:
-        if level <= 0:
-            return True_()
+        assert level >= 0, f"There is no level before level 0."

```

Comment:
I believe this new method has a massive bug
https://discord.com/channels/731205301247803413/1208847391671980143/1269776169184792596

For progressive skills, it should test if you have the current level, not the previous level

Reply:
woops

Reply:
@agilbert1412 should be fixed now